### ISSUE TITLE
Differentiator block

### ISSUE THREAD

**Issue by Tirpitz93** [Hopsan/hopsan#2143]
**Is your improvement request related to a problem? Please describe.**
* I cannot seem to find a differentiator block in the standard lib, yet there is an integrator block. 
* Is there a reason for this, is it named something else?
* I implimented a simple one for myself but I think this should probably be part of the standard lib.

**Describe the solution you'd like**
* A differentiator block in the standard library.
* A Difference block.

**Describe alternatives you've considered**
* I wrote two simple blocks here:
* I will put a PR against `Hopsan/development-libraries`

Difference:
```
        void simulateOneTimestep()
        {
            //Read input variables
            In1 = (*mpIn1);
            Out1 = (*mpOut1);

            //Simulation code


            //Write output variables
            (*mpOut1) = In1 - prev;
            prev = In1;
        }

```
and Differentiator

```
        void simulateOneTimestep() override
        {
            //Read input variables
            In1 = (*mpIn1);
            Out1 = (*mpOut1);

            //Simulation code


            //Write output variables
            (*mpOut1) = (In1 - prev)/mTimestep;
            
            prev = In1;
        }

```

**Comment by Tirpitz93** [comment]
PR Here: https://github.com/Hopsan/development-libraries/pull/6

**Comment by robbr48** [comment]
Good idea. I think the reason we only have an integrator is because integration is more commonly used. Also numerical differentiation can be sensitive to noise, but this problem is more significant for variable step-size simulation. In Hopsan it should work I think.
